packages/contracts/contracts/ChugSplashRegistry.sol

```diff
@@ -113,81 +114,80 @@ contract ChugSplashRegistry is Ownable, Initializable {
-     *                                   denominated as a percentage.
-     */
-    constructor(
-        address _owner,
-        uint256 _ownerBondAmount,
-        uint256 _executionLockTime,
-        uint256 _executorPaymentPercentage,
-        uint256 _protocolPaymentPercentage
-    ) {
-        ownerBondAmount = _ownerBondAmount;
-        executionLockTime = _executionLockTime;
-        executorPaymentPercentage = _executorPaymentPercentage;
-        protocolPaymentPercentage = _protocolPaymentPercentage;
-
+    constructor(address _owner) {
         _transferOwnership(_owner);
     }
 
     /**
-     * @param _executors        Array of executors to add.
+     * @param _executors             Array of executors to add.
+     * @param _initialManagerVersion Initial manager version used for new projects before
+     *                               upgrading to the requested version.
      */
-    function initialize(address[] memory _executors) public initializer {
+    function initialize(
+        address _initialManagerVersion,
+        address[] memory _executors
+    ) public initializer {
         for (uint i = 0; i < _executors.length; i++) {
             executors[_executors[i]] = true;
         }
+
+        initialManagerVersion = _initialManagerVersion;
+        versions[_initialManagerVersion] = true;
     }
 
     /**
      * @notice Claims a new project.
      *
-     * @param _organizationID ID of the new ChugSplash project.
-     * @param _owner     Initial owner for the new project.
+     * @param _organizationID ID of the new ChugSplash organization.
+     * @param _owner     Initial owner for the new organization.
+     * @param _version   Manager version for the new organization.
+     * @param _data      Any data to pass to the ChugSplashManager initalizer.
      */
-    function claim(address _owner, bytes32 _organizationID, bool _allowManagedProposals) public {
+    function claim(
+        bytes32 _organizationID,
+        address _owner,
+        address _version,
+        bytes memory _data
+    ) public {
         require(
             address(projects[msg.sender][_organizationID]) == address(0),
             "ChugSplashRegistry: organization ID already claimed by the caller"
         );
 
-        // Deploy the ChugSplashManager.
+        require(versions[_version] == true, "ChugSplashRegistry: invalid manager version");
+
+        // Deploy the ChugSplashManager proxy and set the implementation to the requested version
         bytes32 salt = keccak256(abi.encode(msg.sender, _organizationID));
-        ChugSplashManager manager = new ChugSplashManager{ salt: salt }(
-            this,
-            executionLockTime,
-            ownerBondAmount,
-            executorPaymentPercentage,
-            protocolPaymentPercentage
+        ChugSplashManagerProxy managerProxy = new ChugSplashManagerProxy{ salt: salt }(
+            address(this),
+            address(this),
+            initialManagerVersion
```

Comment:
I think `initialManagerVersion` should be kept out of the constructor because it could potentially mess up the `ChugSplashManagerProxy`'s Create2 address. For example, say we need to slightly change the bytecode of the initial ChugSplashManager to support an L2. This would change the Create2 address of the ChugSplashManager address, which would in turn cause all of the `ChugSplashManagerProxy` addresses on that chain to be different.

If you agree, I think we can remove `initialManagerVersion` as a state variable